### PR TITLE
Refine kotlin code

### PR DESCRIPTION
**Pull request by curtishd**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

File: .github/workflows/kotlin.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: CI
```
Comment (krahets) [inline .github/workflows/kotlin.yml:1]: Plz add the workflow file in another PR

**Comment by curtishd** [comment]
Sorry, I forgot to switch to different branch for testing workflow.

**Comment by krahets** [comment]
> Sorry, I forgot to switch to different branch for testing workflow.

That's fine. I suggest you double-check all the changes after submitting the PR. It will help eliminate the mistakes.

File: codes/kotlin/chapter_tree/binary_tree_bfs.kt [hunk 2]
```diff
@@ -16,13 +16,15 @@ fun levelOrder(root: TreeNode?): MutableList<Int> {
...
-    while (!queue.isEmpty()) {
-        val node = queue.poll() // 队列出队
-        list.add(node?.value!!) // 保存节点值
-        if (node.left != null) queue.offer(node.left) // 左子节点入队
+    val list = mutableListOf<Int>()
+    while (queue.isNotEmpty()) {
+        val node = queue.poll()      // 队列出队
+        list.add(node?.value!!)      // 保存节点值
+        if (node.left != null)
+            queue.offer(node.left)   // 左子节点入队
```
Comment (krahets) [inline codes/kotlin/chapter_tree/binary_tree_bfs.kt:25]: Remove this empty line

File: codes/kotlin/chapter_tree/array_binary_tree.kt [hunk 3]
```diff
@@ -39,46 +40,51 @@ class ArrayBinaryTree(private val tree: List<Int?>) {
...
 
     /* 层序遍历 */
-    fun levelOrder(): List<Int?> {
-        val res = ArrayList<Int?>()
+    fun levelOrder(): MutableList<Int?> {
+        val res = mutableListOf<Int?>()
         // 直接遍历数组
         for (i in 0..<size()) {
-            if (value(i) != null) res.add(value(i))
+            if (value(i) != null)
```
Comment (krahets) [inline codes/kotlin/chapter_tree/array_binary_tree.kt:47]: Is it good to use `value` as a function name?

**Review by krahets — APPROVED** [review]
Thanks for the impovements!

**Pull request merged**